tests/test_max_number_occurence.py

```diff
@@ -1,13 +1,34 @@
 import unittest
 
-from src.max_number_occurence import generated_dictionary, max_number_occurence
+from src.max_number_occurence import generated_dictionary, max_number_occurrence
 
 
 class GerneratedDictionaryTest(unittest.TestCase):
 
-    def test_equal_type(self):
-        self.assertEqual(type(generated_dictionary()), dict)
+    def test_correct_work(self):
+        test_list = [1, 1, 2]
+        self.assertEqual(generated_dictionary(test_list), {1:2, 2:1})
 
+        second_test_list = [2, 3, 4, 2, 5]
+        self.assertEqual(generated_dictionary(second_test_list), {2:2, 3:1, 4:1, 5:1})
+
+    def test_empty_list(self):
+        test_list = []
+        self.assertEqual(generated_dictionary(test_list), {})
+
+class MaxNumberOccurrenceTest(unittest.TestCase):
+
+    def test_correct_work(self):
+        test_list = [1, 1, 2]
+        self.assertEqual(max_number_occurrence(test_list), (1, 2))
+
+        second_test_list = [7, 3, 4, 6, 7, 9, 8, 6, 6, 4]
+        self.assertEqual(max_number_occurrence(second_test_list), (6, 3))
+
+    def test_empty_list(self):
+        test_list = []
+        with self.assertRaises(IndexError):
```

Comment:
You have to check in `max_number_occurrence` that list is empty, and if it is raise `ValueError` exception.
Right now `IndexError` is raised because your code doesn't work properly with empty lists.